### PR TITLE
[stable/kong] add nginx custom template feature via config map

### PR DESCRIPTION
**Pull request by Mattzr**
#14890 ## What this PR does / why we need it:

To be able to inject custom nginx directive into Kong. This is particularly necessary when using kong enterprise with rbac enabled on the Kong Manager and prometheus plugins is needed.

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [ ] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [ ] Chart Version bumped
- [ ] Variables are documented in the README.md
- [ ] Title of the PR starts with chart name (e.g. `[stable/chart]`)

### CODE REVIEW

File: stable/kong/Chart.yaml
Comment (cpanato) [inline stable/kong/Chart.yaml]: should be 
```suggestion
version: 0.14.0
```

**Comment by hbagdi** [comment]
@cpanato Please do not approve this one and merge in yet. This needs a review from our side as well. Thanks!

**Comment by hbagdi** [comment]
@Mattzr Do you have a use case for exposing the ConfigMap to be configurable?

**Comment by Mattzr** [comment]
@hbagdi - well I personally need it only for the prometheus plugin. However - making the ConfigMap configurable would give any user the ability to inject custom nginx directive depending on their need ( which is native to Kong via KONG_NGINX_HTTP_INCLUDE env var).

**Comment by hbagdi** [comment]
@Mattzr 

I'd say then that let's do incremental changes here.
- Let's add the ConfigMap unconditionally
- Setup the VolumeMount
- Setup the `metrics` port for Kong proxy pods to 9542

We can then tackle the extensible part later.
We are trying to be careful with customization that we introduce into the Helm chart as the possibilities are endless and we would like to add things that makes sense for a wide variety of use case and keep the chart from bloating further.

**Comment by Mattzr** [comment]
@hbagdi done. Please review.

Feel free to change the name of the ConfigMap.

File: stable/kong/templates/controller-deployment.yaml
Comment (hbagdi) [inline stable/kong/templates/controller-deployment.yaml]: This pod will never proxy traffic and hence exposing metrics on this will not give any insights.
Let's remove this.

File: stable/kong/templates/deployment.yaml
Comment (hbagdi) [inline stable/kong/templates/deployment.yaml:232]: In this container, please expose `9542` and call that port `metrics` so that Prometheus can automatically scrape this endpoint.

**Review by hbagdi — CHANGES_REQUESTED** [review]
Some minor tweaks and we should be rolling, thanks!

File: stable/kong/templates/controller-deployment.yaml
Comment (Mattzr) [reply stable/kong/templates/controller-deployment.yaml]: done

File: stable/kong/templates/deployment.yaml
Comment (Mattzr) [reply stable/kong/templates/deployment.yaml:232]: @hba

done removing server block from controller and adding metrics port to Kong pods.

Shall the podAnnotations: default to prometheus annotation then ?

Comment (hbagdi) [reply stable/kong/templates/deployment.yaml:232]: Exposing the port should be enough for some installations while some other installations might need the annotation.
Since pod annotations are already configurable, the user can add those if required.

This is something we can add those too later but I'm being cautious and doing this change incrementally.

Comment (Mattzr) [reply stable/kong/templates/deployment.yaml:232]: ok good point. Since all pods would have the prometheus metrics I thought having the prometheus annotations too would make sense. 

Right ok - let me know if there's anything else missing then.

**Review by hbagdi — APPROVED** [review]
/lgtm

**Comment by hbagdi** [comment]
@Mattzr Thank you for contributing this change!

**Pull request merged**
